packages/zosjobs/__tests__/__system__/GetJobs.system.test.ts

```diff
@@ -179,894 +179,991 @@ describe("Get Jobs - System Tests", () => {
+                    } catch (thrownError) {
+                        error = thrownError;
+                    }
+                    expect(error).toBeDefined();
+                    const trimmedErrorMessage = trimMessage(error.message);
+                    expect(trimmedErrorMessage).toContain("status 400");
+                    expect(trimmedErrorMessage).toContain(job.jobid);
+                    testEnvironment.resources.jobs.push(job);
+                });
+            });
+
+            describe("download JCL", () => {
+                it("should be able to get jcl from a job that was submitted", async () => {
+                    const job = await SubmitJobs.submitJcl(REAL_SESSION, JCL);
+                    const jcl = await GetJobs.getJclForJob(REAL_SESSION, job);
+                    expect(jcl).toContain("EXEC PGM=IEFBR14");
+                    expect(jcl).toContain("JOB");
+                    testEnvironment.resources.jobs.push(job);
+                });
+            });
         });
     });
-});
 
+    describe("Encoded System Tests", () => {
 
-describe("Get Jobs - System Tests - Encoded", () => {
+        beforeAll(async () => {
+            const REAL_SESSION = await TestEnvironment.createSession();
 
-    beforeAll(async () => {
-        testEnvironment = await TestEnvironment.setUp({
-            testName: "zos_get_jobs_encoded"
-        });
-        defaultSystem = testEnvironment.systemTestProperties;
+            testEnvironment = await TestEnvironment.setUp({
+                testName: "zos_get_jobs_encoded"
+            }, REAL_SESSION);
+            defaultSystem = testEnvironment.systemTestProperties;
+
+            INVALID_SESSION = new Session({
+                user: "fakeuser",
+                password: "fake",
+                hostname: defaultSystem.zosmf.host,
+                port: defaultSystem.zosmf.port,
+                type: "basic",
+                rejectUnauthorized: false
+            });
 
-        REAL_SESSION = TestEnvironment.createZosmfSession(testEnvironment);
+            ACCOUNT = defaultSystem.tso.account;
+            MONITOR_JOB_NAME = REAL_SESSION.ISession.user?.toUpperCase().substring(0, 5) + "#G";
 
-        INVALID_SESSION = new Session({
-            user: "fakeuser",
-            password: "fake",
-            hostname: defaultSystem.zosmf.host,
-            port: defaultSystem.zosmf.port,
-            type: "basic",
-            rejectUnauthorized: false
+            JOBCLASS = testEnvironment.systemTestProperties.zosjobs.jobclass;
+            SYSAFF = testEnvironment.systemTestProperties.zosjobs.sysaff;
+
+            // TODO: What string goes in the removed section?
+            JCL =
+                "//" + MONITOR_JOB_NAME + " JOB '" + ACCOUNT + "',CLASS=" + JOBCLASS + "\n" +
+                "//IEFBR14 EXEC PGM=IEFBR14"; // GetJobs
+        });
+        afterAll(async () => {
+            await TestEnvironment.cleanUp(testEnvironment);
```

Comment:
Referring back to the previous comment, do we think we might be trying to clean up the same resources over an over? 😋